ipn/ipnstate/ipnstate.go

```diff
@@ -216,6 +216,8 @@ type PeerStatus struct {
 
 	// TailscaleIPs are the IP addresses assigned to the node.
 	TailscaleIPs []netip.Addr
+	// AllowedIPs are IP addresses allowed to route to this node.
+	AllowedIPs *views.Slice[netip.Prefix] `json:",omitempty"`
```

Comment:
I don't think this really needs to be a pointer?  We seem to be inconsistent in whether we typically use pointers to these slices, but I think we generally don't need to.  The zero value would just be a struct with a nil internal slice.

Reply:
We use the pointer for `PrimaryRoutes` and `Tags` on the `PeerStatus` already, so did this for consistency with those. I think it will make a difference with `json:",omitempty"` - only actually omitted if nil.

Reply:
yeah what @soniaappasamy said. Using the pointer makes it possible for us to omitempty. However, in the case of `AllowedIPs` I don't really expect it to ever be empty. It's fine to do it this way though.